OpenEdXMobile/src/main/java/org/edx/mobile/view/LoginActivity.java

```diff
@@ -235,24 +235,32 @@ public String getEmail() {
         return activityLoginBinding.emailEt.getText().toString().trim();
     }
 
-    private static final int RESET_PASSWORD_REQUEST_CODE = 0;
-
     private void showResetPasswordDialog() {
-        startActivityForResult(ResetPasswordActivity.newIntent(getEmail()), RESET_PASSWORD_REQUEST_CODE);
+        resetPasswordAlert = ResetPasswordAlertDialogFragment.newInstance(this, getEmail());
+        resetPasswordAlert.setListener(this);
```

Comment:
> I also thought about modifying LoginActivity to put the methods right there, then calling it from the fragment (as you suggested). The issue that I had with that is that it didn't seem very modular. By modifying the activity directly, we are intrinsically tying the ResetPasswordDialog with LoginActivity and that didn't feel right to me.

My suggestion was to actually use a callback interface (though my preferred one would only define the `onActivityResult()` method as public), so it isn't tying the DialogFragment to any particular Activity.

> Initially I had the same reservations as you, but then I remembered that if LoginActivity is re-instantiated, the screen would be reset anyway. There isn't any code to make sure that any popup retains its state after being re-instantiated (and if we need to do that we would handle the listener there).

But _this_ is not very modular 😜 even if it works in this particular case.

> The fragment does have its own lifecycle, but it is dependent on the enclosing activity (in other words, as long as activity is alive, we shouldn't worry about dangling pointers).

Not really; it can be dismissed at any point without any reference to the Activity. You have set an `onDismissed()` callback that sets the reference to `null`, but really you don't need to have any reference to it at all if it simply takes care of dismissing itself at the proper time instead of leaving it the Activity callback (which it invokes itself as well).

If you _did_ need to obtain a reference to a Fragment, then you should use the FragmentManager for it, which manages the Fragment sessions/lifecycles, instead of trying to manage it yourself.